docs/input_processing.md

```diff
@@ -1,25 +1,13 @@
-As of now two inputs are supported:
-- JSON Schema Draft 7
-- AsyncAPI version 2.0.0
+# Input process
 
-## Internal model representation
+The [InputProcessor](../src/processors/InputProcessor.ts) is our main point of entry for processing input data. It uses the defined input processors (`AsyncAPIInputProcessor`, `JsonSchemaInputProcessor`, ...) by first calling `shouldProcess` function for each input processor to figure out if the input data should be processed by the respective processor. If it should the `process` function is then called. It uses a first-come, first-serve principle, where the first processor that accepts the input process's it. If no processor accepts the input it defaults to `JsonSchemaInputProcessor`. 
 
-![Class diagram](./assets/class_diagram.png)
+The `process` function are expected to return `CommonInputModel` which is a wrapper for the core data representation of `CommonModel`. This is done to ensure we can return multiple models for any input to allow for references, inheritance etc. 
 
-As seen on the class diagram the `InputProcessor` is our main point of entry for processing input data. 
-
-It uses the defined input processors (`AsyncAPIInputProcessor`, `JsonSchemaInputProcessor`, ...) by first calling `shouldProcess` function of each and if the function returns true it calls the `process` function. 
-
-If no processes returns true it defaults to `JsonSchemaInputProcessor`. 
-
-The `process` function are expected to return `CommonInputModel` which is a wrapper for the core data representation of `CommonModel`. 
-
-This is done to ensure we can return multiple models for any input to allow for references, inheritance etc. 
-
-As said the core internal representation of a data model is `CommonModel`. This contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represent data definition. The explanation for the `CommonModel` properties can be found [here](../API.md#CommonModel).
+As said the core internal representation of a data model is `CommonModel`. It contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represents the data definition for the model. The explanation for the `CommonModel` properties can be found [here](../API.md##CommonModel).
```

Comment:
```suggestion
As said the core internal representation of a data model is `CommonModel`. It contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represents the data definition for the model. The explanation for the `CommonModel` properties can be found [here](../API.md#CommonModel).
```

Duplicated `#` and due to this link doesn't work.

Reply:
Hmm.... It was actually deliberate since I wanted it to link to the model itself, not sure why it is not working now...

Reply:
For me it's not working 😄